src/main/Core/ImageGenerator/LinuxIconExtractor.ts

```diff
@@ -0,0 +1,354 @@
+        let fileName = this.findIconHelper(icon, size, scale, theme);
+        if (fileName) return fileName;
+
+        fileName = this.findIconHelper(icon, size, scale, "hicolor");
+        if (fileName) return fileName;
+
+        return this.lookupFallbackIcon(icon);
+    }
+
+    private findIconHelper(icon: string, size: number, scale: number, theme: string): string {
+        if (!this.searchCache.has(theme)) return "";
+        let fileName = this.lookupIcon(icon, size, scale, theme);
+        if (fileName) return fileName;
+
+        for (const themeIndex of this.searchCache.get(theme)) {
+            if (themeIndex.parents) {
+                for (const parent of themeIndex.parents) {
+                    fileName = this.findIconHelper(icon, size, scale, parent);
+                    if (fileName) return fileName;
+                }
+            }
+        }
+        return "";
+    }
+
+    private lookupIcon(iconName: string, size: number, scale: number, theme: string): string {
+        for (const themeIndex of this.searchCache.get(theme)) {
+            for (const subdir of themeIndex.subdirectories) {
+                for (const extension of ["png", "svg", "xpm"]) {
+                    if (
+                        themeIndex.subdirData.has(subdir) &&
+                        this.directoryMatchesSize(themeIndex.subdirData.get(subdir), size, scale)
+                    ) {
+                        const filename = `${join(themeIndex.path, subdir, iconName)}.${extension}`;
+                        if (this.fileSystemUtility.existsSync(filename)) return filename;
+                    }
+                }
+            }
+        }
+        let minimalSize = Number.MAX_SAFE_INTEGER;
+        let closestFilename = "";
+        for (const themeIndex of this.searchCache.get(theme)) {
+            for (const subdir of themeIndex.subdirectories) {
+                for (const extension of ["png", "svg", "xpm"]) {
+                    if (!themeIndex.subdirData.has(subdir)) continue;
+                    const filename = `${join(themeIndex.path, subdir, iconName)}.${extension}`;
+                    const dist = this.directorySizeDistance(themeIndex.subdirData.get(subdir), size, scale);
+                    if (this.fileSystemUtility.existsSync(filename) && dist < minimalSize) {
+                        closestFilename = filename;
+                        minimalSize = dist;
+                    }
+                }
+            }
+        }
+        return closestFilename;
+    }
+
+    private lookupFallbackIcon(iconName: string): string {
+        for (const directory of this.baseDirectories) {
+            for (const extension of ["png", "svg", "xpm"]) {
+                const fileName = `${join(directory, iconName)}.${extension}`;
+                if (this.fileSystemUtility.existsSync(fileName)) return fileName;
+            }
+        }
+        if (this.fileSystemUtility.existsSync(iconName)) return iconName;
+        return "";
```

Comment:
I suggest avoiding returning an empty string. Why not return `undefined` instead? In my opinion that would make it much more clear that no icon could be found/generated.